File: 백준/Silver/1181. 단어 정렬/단어 정렬.py

# 길이가 짧은 것부터
# 길이가 같다면 사전순
# 중복된 단어는 하나만 남기고 제거
word = set()

N = int(input())
for _ in range(N):
    word.add(input())

word = list(word)
word.sort(key=lambda x: (len(x), x))

print(*word, sep="\n")

File: 백준/Silver/1181. 단어 정렬/단어 정렬.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

function compare(a, b) {
  return a.length - b.length;
}

let n = Number(input[0]);
let strSet = new Set();

for (let i = 1; i <= n; i++) {
  strSet.add(input[i]);
}

let arr = [...strSet];
arr.sort();
arr.sort(compare);
console.log(arr.join('\n'));
